src/main/kotlin/com/jaoafa/vcspeaker/voicetext/textreplacers/UrlReplacer.kt

```diff
@@ -0,0 +1,455 @@
+        "ps1" to "PowerShellスクリプトファイル",
+        "bat" to "バッチファイル",
+        "cmd" to "コマンドファイル",
+        "sh" to "シェルスクリプトファイル",
+        "js" to "JavaScriptファイル",
+        "htm" to "HTMLファイル",
+        "py" to "Pythonファイル",
+        "rb" to "Rubyファイル",
+        "java" to "Javaファイル",
+        "cpp" to "Cプラスプラスファイル",
+        "cs" to "C#ファイル",
+        "kt" to "Kotlinファイル",
+        "rs" to "Rustファイル",
+        "yml" to "ヤメルファイル",
+        "yaml" to "ヤメルファイル",
+        "conf" to "設定ファイル",
+        "config" to "設定ファイル",
+        "md" to "Markdownファイル",
+    )
+
+
+    override suspend fun replace(text: String, guildId: Snowflake): String {
+        suspend fun replaceUrl(vararg replacers: suspend (String, Snowflake) -> String) =
+            replacers.fold(text) { replacedText, replacer ->
+                replacer(replacedText, guildId)
+            }
+
+        return replaceUrl(
+            ::replaceMessageUrl,
+            ::replaceChannelUrl,
+            ::replaceEventDirectUrl,
+            ::replaceEventInviteUrl,
+            ::replaceTweetUrl,
+            ::replaceInviteUrl,
+            ::replaceUrlToTitle,
+            ::replaceUrl,
+        )
+    }
+
+    /**
+     * チャンネルタイプに応じて、読み上げるチャンネル種別テキストを返します。
+     */
+    private fun getChannelTypeText(channel: GuildChannel): String {
+        return when (channel.type) {
+            is ChannelType.GuildText -> "テキストチャンネル"
+            is ChannelType.GuildVoice -> "ボイスチャンネル"
+            is ChannelType.GuildCategory -> "カテゴリ"
+            is ChannelType.GuildNews -> "ニュースチャンネル"
+            else -> channel.type.toString() + "チャンネル"
+        }
+    }
+
+    /**
+     * チャンネルIDをもとに、チャンネルを取得します。スレッドの場合は、親チャンネルを取得します。
+     */
+    private suspend fun getChannel(guild: Guild, channelId: Snowflake): GuildChannel? {
+        val channel = guild.getChannelOrNull(channelId) ?: return null
+
+        val isThread =
+            channel.type == ChannelType.PrivateThread || channel.type == ChannelType.PublicGuildThread || channel.type == ChannelType.PublicNewsThread
+        if (isThread) {
+            val thread = channel.asChannelOf<ThreadChannel>()
+            return thread.parent.asChannel()
+        }
+
+        return channel
+    }
+
+    /**
+     * チャンネルIDをもとに、スレッドを取得します。スレッドでない場合はnullを返します。
+     */
+    private suspend fun getThread(guild: Guild, channelId: Snowflake): ThreadChannel? {
+        val channel = guild.getChannelOrNull(channelId) ?: return null
+
+        val isThread =
+            channel.type == ChannelType.PrivateThread || channel.type == ChannelType.PublicGuildThread || channel.type == ChannelType.PublicNewsThread
+        if (isThread) {
+            return channel.asChannelOf<ThreadChannel>()
+        }
+
+        return null
+    }
+
+    /**
+     * 招待リンクをもとに、招待の情報を取得します。
+     */
+    private suspend fun getInvite(inviteId: String, eventId: Snowflake? = null): DiscordInvite? {
+        val url = "https://discord.com/api/invites/${inviteId}"
```

Comment:
```suggestion
        val url = "https://discord.com/api/invites/$inviteId"
```
`.` も ` ` も入らない変数と式は `{}` を省略できます